app/celery/tasks.py

```diff
@@ -314,7 +315,7 @@ def save_letter(
             service=service,
             personalisation=notification["personalisation"],
             notification_type=LETTER_TYPE,
-            api_key_id=None,
+            api_key_id=notification.get("api_key"),
```

Comment:
Hmm @Moro-Code made similar code changes in his PR, although he defaults to `None`. Both of your changes might clash. You might want to sync together before we merge one or the other.

Reply:
It doesn't hurt to explicitly add a default. Can we just do that ? job_id seems to do the same thing `job_id=notification.get("job", None)`